docs/src/modules/java-protobuf/pages/actions-publishing-subscribing.adoc

```diff
@@ -233,4 +233,114 @@ Referencing environment variables is done with the syntax `$\{VAR_NAME}` in the
 
 Note that if changing the `topic` name after it has once been deployed for an event consumer means the consumer will start over from the beginning of the topic.
 
-See https://docs.kalix.io/kalix/kalix_services_deploy.html[kalix service deploy] for details on how to set environment variables when deploying a service.
\ No newline at end of file
+See https://docs.kalix.io/kalix/kalix_services_deploy.html[kalix service deploy] for details on how to set environment variables when deploying a service.
+
+
+== Testing the Integration
+
+When a Kalix service relies on a broker, it might be useful to use integration tests to assert that those boundaries work as intended. For such scenarios, you can either:
+
+* Use TestKit's mocked topic:
+** this offers a general API to inject messages into topics or read the messages written to another topic, regardless of the specific broker integration you have configured.
+* Run an external broker instance:
+** if you're interested in running your integration tests against a real instance, you need to provide the broker instance yourself by running it in a separate process in your local setup and make sure to disable the use of TestKit's test broker. Currently, **the only external broker supported in integration tests is Google PubSub Emulator.**
```

Comment:
AFAIR, it is indeed impossible to run integration tests against Kafka at the moment becuase the proxy container run from the integration tests supports either the mock broker or the google pub sub (before the eventing testkit it would only support google pub sub). This is somehting we can (and should) easily change... it's just a matter of allowing that proxy to be configured for kafka as well (meaning we set a differnet eventing support method and a different port by default). But at the moment, I don't think we have configs for that.